### ISSUE TITLE
List active "markets" of the MetaDEx

### ISSUE THREAD

**Issue by dexX7** [OmniLayer/omnicore#201]
While there is `"omni_getorderbook"` to retrieve active orders, there is no convenient way to learn which currency pairs are actually traded.

As workaround it's possible to iterate over `"omni_listproperties"` and then check, whether `"omni_getorderbook"` returns any results..

A straightforward solution could be to add a new RPC, such as `"omni_getactivemarkets"`, and then iterate over all in-memory trades, push the properties for sale, and properties desired, to a set, and, depending on whether we want more details, either return the content of the set, or use the set to also retrieve property names etc., and return more detailed information in an array of objects.

@zathras-crypto: if I recall correctly, then you tackled this earlier, but ditched the approach, because we shouldn't assume currency pairs are always "token/MSC", but may as well be "token/token" in the future?

**Comment by zathras-crypto** [comment]
> @zathras-crypto: if I recall correctly, then you tackled this earlier, but ditched the approach, because we shouldn't assume currency pairs are always "token/MSC", but may as well be "token/token" in the future?

That's correct, I didn't want to see us have a bunch of RPC calls that were restricted to just "token/MSC" and have a whole new set of RPCs for "token/token" when we allow non-MSC MetaDEx trading as I thought it would bloat the stack and force us to duplicate code.  It's a shame because the whole "market" idea really simplified things short term hehe...

> While there is "omni_getorderbook" to retrieve active orders, there is no convenient way to learn which currency pairs are actually traded.

> As workaround it's possible to iterate over "omni_listproperties" and then check, whether "omni_getorderbook" returns any results..

I'm not actually 100% certain on what you're trying to achieve here dude (but I am probably missing the point hehe).  Would wildcarding on ```omni_getorderbook``` address the requirement?  (eg allow to pull all open trades across all markets instead of all open trades *for a specific pair*?)

**Comment by dexX7** [comment]
> That's correct, I didn't want to see us have a bunch of RPC calls that were restricted to just "token/MSC" 

Yes, I agree with this course.

> I'm not actually 100% certain on what you're trying to achieve here dude

Hehe, so the use case basically was to have an external website listing open orders, and I thought about how one could get an idea, which markets, or currency pairs, are available to be listed.

Using a wildcard for `"omni_getorderbook"` would work, but this is pretty extreme imho.

In it's most basic form, I think somethink like the following would already be useful:
```js
[{
    "propertyidforsale": 1,
    "propertyiddesired": 3
}, {
    "propertyidforsale": 3,
    "propertyiddesired": 1
}, {
    ...
}]
```
On the other hand, a "pair" of currencies has always "two sides", even though there may be no trades for one side (i.e. there is always [(1, 3), (3, 1)], which raises, at least for me, the question, whether "inverted pairs" might be included in the results, even though there are no trades.

From an user's perspective, let's say we'd like to reconstruct all open trades (which would be the same as `"omni_getorderbook"` with wildcard, but just for the sake of an example), then I'd imagine one would use `"omni_listtradepairs"` (or so) and iterate over the results, and fetch the active trades for each pair via `"omni_getorderbook <id> <id>"`.

**Comment by zathras-crypto** [comment]
> I'd imagine one would use "omni_listtradepairs"

Sure sure, OK thanks for the explanation - in that case sure I don't see any reason we can't have such an RPC call - though a minor nit; could we call it ```omni_listactivetradepairs``` since technically any pair is a trade pair, but not necessarily active (ie with live trades).

> which raises, at least for me, the question, whether "inverted pairs" might be included in the results, even though there are no trades

My initial thought is not to include inverted pairs that don't have any matches, since that pair doesn't have any live trades going on.  It may be unintuitive to request the pairs with live trades, and then see that some of the returned pairs don't actually have any trades.
